### ISSUE TITLE
C# refactoring - slightly confusing convertion from 'if' construct to 'switch'

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#40168]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/829431/c-refactoring-slightly-confusing-convertion-from-i.html)._

---
My VS version is 16.3.10.

When using the quick action "Convert 'if' construct to 'switch'", the code transformation, although apparently correct, looks slightly confusing.

Original code:

```
            if ((a == 2 && b == 14) || (a == 1 && b == 15))
                Console.WriteLine("Complicated case");
            else if (a == 3)
                Console.WriteLine("3");
            else if (a == 0)
                Console.WriteLine("Zero");
            else
                Console.WriteLine("Another case");
```
Expected result:

```
            switch (a)
            {
                case 2 when b == 14:
                case 1 when b == 15:
                    Console.WriteLine("Complicated case");
                    break;
                case 3:
                    Console.WriteLine("3");
                    break;
                case 0:
                    Console.WriteLine("Zero");
                    break;
                default:
                    Console.WriteLine("Another case");
                    break;
            }
```

Actual result:

```
            switch (a)
            {
                case 2 when b == 14 || (a == 1 && b == 15):
                case 1 when b == 15 || (a == 1 && b == 15):
                    Console.WriteLine("Complicated case");
                    break;
                case 3:
                    Console.WriteLine("3");
                    break;
                case 0:
                    Console.WriteLine("Zero");
                    break;
                default:
                    Console.WriteLine("Another case");
                    break;
            }
```

---
#### Visual Studio Feedback System on 11/22/2019, 01:06 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by alrz** [comment]
Verified that this is fixed in 16.5 preview 1

**Comment by jinujoseph** [comment]
Thanks @alrz
